launcher/LoggedProcess.cpp

```diff
@@ -66,13 +66,12 @@ QStringList LoggedProcess::reprocess(const QByteArray& data, QTextDecoder& decod
     }
 
 #if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
-    auto lines = str.remove(QChar::CarriageReturn).split(QChar::LineFeed, QString::SkipEmptyParts);
+    auto lines = str.remove(QChar::CarriageReturn).split(QChar::LineFeed);
 #else
-    auto lines = str.remove(QChar::CarriageReturn).split(QChar::LineFeed, Qt::SkipEmptyParts);
+    auto lines = str.remove(QChar::CarriageReturn).split(QChar::LineFeed);
 #endif
 
-    if (!str.endsWith(QChar::LineFeed))
-        m_leftover_line = lines.takeLast();
+    m_leftover_line = lines.takeLast();
```

Comment:
This change might cause a regression like #930

Reply:
I don't think so.. the only reason for having the conditional before is that with `SkipEmptyParts` the last split part is always a complete line if the data ends in `\n`, whereas without `SkipEmptyParts` the last split part is empty, so we can unconditionally take it as the leftover line (even though it is empty). For partial line data we still store the leftover line as before.